### DISCUSSION TITLE
how long zfs events are retained

### DISCUSSION THREAD

**Discussion by mokank** [openzfs/zfs#13682] [Q&A]
Hi

I would like to retain the zfs events forever, do we have any properties to set this value.
Could someone guide please

also i dont see any properties been available to retain the zfs events loner.

[05:23:02] labtest:~ # zfs get all test
NAME             PROPERTY              VALUE                  SOURCE
Test  type                  filesystem             -
test  creation              Fri Nov 16 15:40 2018  -
test  used                  114G                   -
test  available             272G                   -
test  referenced            96K                    -
test  compressratio         2.14x                  -
test  mounted               yes                    -
test  quota                 none                   default
test  reservation           none                   default
test  recordsize            128K                   default
test  mountpoint            /test       default
test  sharenfs              off                    default
test  checksum              on                     default
test  compression           lz4                    local
test  atime                 on                     default
test  devices               on                     default
test  exec                  on                     default
test  setuid                on                     default
test  readonly              off                    default
test  zoned                 off                    default
test  snapdir               hidden                 default
test  aclinherit            restricted             default
test  createtxg             1                      -
test  canmount              on                     default
test  xattr                 on                     default
test  copies                1                      default
test  version               5                      -
test  utf8only              off                    -
test  normalization         none                   -
test  casesensitivity       sensitive              -
test  vscan                 off                    default
test  nbmand                off                    default
test  sharesmb              off                    default
test  refquota              none                   default
test  refreservation        none                   default
test  guid                  15252145084986466622   -
test  primarycache          all                    default
test  secondarycache        all                    default
test  usedbysnapshots       0B                     -
test  usedbydataset         96K                    -
test  usedbychildren        114G                   -
test  usedbyrefreservation  0B                     -
test  logbias               latency                default
test  dedup                 off                    default
test  mlslabel              none                   default
test  sync                  standard               default
test  dnodesize             legacy                 default
test  refcompressratio      1.00x                  -
test  written               96K                    -
test  logicalused           243G                   -
test  logicalreferenced     34.5K                  -
test  volmode               default                default
test  filesystem_limit      none                   default
test  snapshot_limit        none                   default
test  filesystem_count      none                   default
test  snapshot_count        none                   default
test  snapdev               hidden                 default
test  acltype               off                    default
test  context               none                   default
test  fscontext             none                   default
test  defcontext            none                   default
test  rootcontext           none                   default
test  relatime              off                    default
test  redundant_metadata    all                    default
test  overlay               off                    default

**Comment by rincebrain** [comment]
It's ambiguous what you mean by "zfs events", but the contents of `zpool events` is not currently persistent past reboot on Linux or FreeBSD, and the buffer that reads from is finite in a way that at least on Linux I believe is hardcoded in addition to that. 

Patches welcome, but it seems so far nobody has been bothered enough by this to change it.

I think most people who want somewhat indefinitely persistent `zpool events` data add a zedlet that logs the incoming events somewhere for them.
